src/shared/infra/repositories/product_repository_dynamo.py

```diff
@@ -89,29 +89,27 @@ def delete_product(self, product_id: str, restaurant: RESTAURANT) -> Product:
         return ProductDynamoDTO.from_dynamo(delete_product["Products"]).to_entity()
 
     def update_product(self, restaurant: RESTAURANT, product_id: str, new_available: bool = None, new_price: float = None, new_name: str = None, new_description: str = None, new_prepare_time: int = None, new_meal_type: MEAL_TYPE = None, new_photo: str = None, new_last_update: int = None) -> Product:
-        if self.get_product(product_id=product_id, restaurant=restaurant) == None: 
+
+        product_to_update = self.get_product(product_id=product_id, restaurant=restaurant)
+
+        if product_to_update is None: 
             return None
-        
-        update_product = Product(
-            restaurant=restaurant,
-            product_id=product_id,
-            available=new_available,
-            price=new_price,
-            name=new_name,
-            description=new_description,
-            prepare_time=new_prepare_time,
-            meal_type=new_meal_type,
-            photo=new_photo,
-            last_update=new_last_update
-            )
-            
-        update_product_dto = ProductDynamoDTO.from_entity(product=update_product).to_dynamo()
-
-        response = self.dynamo.hard_update_item(
+
+        response = self.dynamo.update_item(
             partition_key=self.partition_key_format(restaurant=restaurant),
             sort_key=self.sort_key_format(product_id=product_id),
-            item=update_product_dto,
-            is_decimal=True
-        )
+            update_dict={
```

Comment:
isso aqui vai dar errado, se vc 
![image](https://github.com/Maua-Dev/mf_mss_product/assets/81604963/4d1d3f0f-d303-4186-a4fb-2b0c06c746eb)

ou no .value

Reply:
além disso, eu acho que vc tem que tirar o que for None, senão o dynamo vai dar update nisso, testem melhor isso pra ter ceteza

Reply:
arrumado